aggregator/election_mgr.go

```diff
@@ -504,6 +508,20 @@ func (mgr *electionManager) verifyPendingFollower(watch watch.Watch) {
 				mgr.logError("leader has not changed", errLeaderNotChanged)
 				return errLeaderNotChanged
 			}
+			_, p, err := mgr.placementManager.Placement()
+			if err != nil {
+				mgr.metrics.verifyPlacementErrors.Inc(1)
+				mgr.logError("error getting placement", err)
+				return err
+			}
+			_, exist := p.Instance(leader)
+			if !exist {
+				mgr.metrics.verifyLeaderNotInPlacement.Inc(1)
+				err := fmt.Errorf("received invalid leader value: [%s], which is not available in placement", leader)
+				mgr.logger.Error(err.Error())
+				return err
+			}
+			mgr.logger.Infof("found valid new leader: [%s] for the campaign", leader)
```

Comment:
Do you need check if the current instance and the leader instance are in the same group?